fabric-testmods/java/net/fabricmc/fabric/particles/ParticleModClient.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.particles;
+
+import net.fabricmc.api.ClientModInitializer;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.api.ModInitializer;
+import net.fabricmc.fabric.api.particles.ParticleFactoryRegistry;
+import net.fabricmc.fabric.api.particles.FabricParticleTypes;
+import net.fabricmc.fabric.api.particles.FabricSpriteProvider;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.particle.AnimatedParticle;
+import net.minecraft.client.particle.Particle;
+import net.minecraft.client.particle.ParticleFactory;
+import net.minecraft.client.particle.ParticleTextureSheet;
+import net.minecraft.client.particle.SpriteBillboardParticle;
+import net.minecraft.client.particle.SpriteProvider;
+import net.minecraft.item.Items;
+import net.minecraft.particle.DefaultParticleType;
+import net.minecraft.particle.ParticleEffect;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.World;
+
+public class ParticleModClient implements ClientModInitializer, ModInitializer {
```

Comment:
This and other test mod files should be removed from the PR and used locally only. Don't want to ship them with the API, after all.

Reply:
It's in the fabric-testmods module as an example (something people badgered me about including with my previous PR).

There _is_ a lot more garbage involved with actually getting the test mod to run, that I've avoided committing.

Reply:
Ah, okay. My bad, I missed that in the path.